orders/views.py

```diff
@@ -3,52 +3,68 @@
 from django.views import View
 from django.http  import JsonResponse
 
-from .models         import Order
-from users.models    import User
-from products.models import Product, Image
+from .models         import Order, CartItem, Status
+from products.models import Product
 from users.utils     import UserInfoDeco
 
 class CartView(View):
     @UserInfoDeco
-    def post(self, request): # 제품상세에서 장바구니 눌렀을 때, 우선 DB에 저장
-        try:
-            user     = request.user # token 유저정보
-            data     = json.loads(request.body)
-            product  = Product.objects.get(id=data['productId'])
-            size     = data['size']
-            quantity = data['quantity']
-
-            if Order.objects.filter(user_id=user.id, product_id=product.id, size=size).exists(): # 장바구니에 이미 같은 제품이 들어가있다면 갯수만 추가
-                user_order           = Order.objects.get(user_id=user.id, product_id=product.id, size=size)
-                user_order.quantity += int(quantity)
-                user_order.save()
-                return JsonResponse({"message":"장바구니 업데이트 성공"}, status=200)
-            else: # 아니라면 장바구니 생성
-                Order.objects.create(
-                    user_id    = user.id,
-                    product_id = product.id,
-                    size       = size,
-                    quantity   = quantity
-                    )
-                return JsonResponse({"message":"장바구니 생성 성공"}, status=200)
-        except Product.DoesNotExist:
-            return JsonResponse({"message":"잘못된 product"}, status=500)
+    def post(self, request):
+        user     = request.user
+        data     = json.loads(request.body)
+        size     = data['size']
+        quantity = data['quantity']
+        product  = Product.objects.get(id=data['productId'])
+        status   = Status.objects.get(id=1)
+        cart     = Order.objects.filter(user_id=user.id, status=status).first()
+        
+        if not cart:
+            Order.objects.create(user_id=user.id, status=status)  # 유저의 테이블 없다면 생성
+
+        if CartItem.objects.filter(order_id=cart.id, product_id=product.id, size=size).exists(): # 이미 CartItem에 동일 제품이 있다면 갯추만 증가
+            user_order           = CartItem.objects.get(order_id=cart.id, product_id=product.id, size=size)
+            user_order.quantity += int(quantity)
+            user_order.save()
+            return JsonResponse({"message":"cart update success"}, status=200)
+
+        CartItem.objects.create(
+            order_id   = cart.id,
+            product_id = product.id,
+            size       = size,
+            quantity   = quantity
+            )
+        return JsonResponse({"message":"cart add success"}, status=200)
```

Comment:
get_or_create 에 대해 알아보시겠어요?

```suggestion
        Order, created = Order.objects.get_or_create(user_id=user.id, status=status)
        
        if created:
            Order.objects.create(user_id=user.id, status=status)
            OrderItem.objects.create( ~~~~~~)
            return success
        order_item = OrderItem.objects.get(order_id = ~~~~~)
        order_item.quanity += ~~~~~~
        return success
            
        
```